lib/src/layer/marker_layer.dart

```diff
@@ -110,44 +110,34 @@ class MarkerLayer extends StatelessWidget {
   MarkerLayer(this.markerOpts, this.map, this.stream)
       : super(key: markerOpts.key);
 
-  bool _boundsContainsMarker(Marker marker) {
-    var pixelPoint = map.project(marker.point);
-
-    final width = marker.width - marker.anchor.left;
-    final height = marker.height - marker.anchor.top;
-
-    var sw = CustomPoint(pixelPoint.x + width, pixelPoint.y - height);
-    var ne = CustomPoint(pixelPoint.x - width, pixelPoint.y + height);
-    return map.pixelBounds.containsPartialBounds(Bounds(sw, ne));
-  }
-
   @override
   Widget build(BuildContext context) {
     return StreamBuilder<int>(
       stream: stream, // a Stream<int> or null
       builder: (BuildContext context, AsyncSnapshot<int> snapshot) {
         var markers = <Widget>[];
-        for (var markerOpt in markerOpts.markers) {
-          var pos = map.project(markerOpt.point);
-          pos = pos.multiplyBy(map.getZoomScale(map.zoom, map.zoom)) -
-              map.getPixelOrigin();
+        for (var marker in markerOpts.markers) {
+          var pxPoint = map.project(marker.point);
 
-          var pixelPosX =
-              (pos.x - (markerOpt.width - markerOpt.anchor.left)).toDouble();
-          var pixelPosY =
-              (pos.y - (markerOpt.height - markerOpt.anchor.top)).toDouble();
+          final width = marker.width - marker.anchor.left;
+          final height = marker.height - marker.anchor.top;
+          var sw = CustomPoint(pxPoint.x + width, pxPoint.y - height);
+          var ne = CustomPoint(pxPoint.x - width, pxPoint.y + height);
 
-          if (!_boundsContainsMarker(markerOpt)) {
+          if (!map.pixelBounds.containsPartialBounds(Bounds(sw, ne))) {
             continue;
           }
 
+          final pos = pxPoint.multiplyBy(map.getZoomScale(map.zoom, map.zoom)) -
```

Comment:
Like... what the hell is that? `map.getZoomScale(map.zoom, map.zoom)` will always return a `1.0`

# :thinking:

Reply:
I agree, it looks a bit strange and unesserary @johnpryan? I see the same in all layer types, not just in marker_layer.
![image](https://user-images.githubusercontent.com/866528/111533827-83f22d80-8767-11eb-833f-919076be3c2c.png)
Only real use is when layer and map is at different zoom levels, as is the case in tile_layer.